unit_test/test_spdm_responder/algorithms.c

```diff
@@ -2166,6 +2165,61 @@ void libspdm_test_responder_algorithms_case23(void **state)
     spdm_context->local_context.capability.flags |= SPDM_GET_CAPABILITIES_REQUEST_FLAGS_PSK_CAP;
     spdm_context->connection_info.capability.flags |= SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_PSK_CAP;
 
+    /* Sub Case 1: other_params_support set OpaqueDataFmt1 */
+    m_libspdm_negotiate_algorithm_request18.spdm_request_version10.other_params_support =
+        SPDM_ALGORITHMS_OPAQUE_DATA_FORMAT_1;
+    spdm_context->local_context.algorithm.other_params_support =
+        SPDM_ALGORITHMS_OPAQUE_DATA_FORMAT_1;
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES;
+    libspdm_reset_message_a(spdm_context);
+
+    response_size = sizeof(response);
+    status = libspdm_get_response_algorithms (spdm_context,
+                                              m_libspdm_negotiate_algorithm_request18_size,
+                                              &m_libspdm_negotiate_algorithm_request18,
+                                              &response_size,
+                                              response);
+    assert_int_equal(status, LIBSPDM_STATUS_SUCCESS);
+    assert_int_equal(response_size,
+                     sizeof(spdm_algorithms_response_t) + 4*
+                     sizeof(spdm_negotiate_algorithms_common_struct_table_t));
+    spdm_response = (void *)response;
+    assert_int_equal(spdm_response->header.spdm_version, SPDM_MESSAGE_VERSION_12);
+    assert_int_equal(spdm_response->header.request_response_code, SPDM_ALGORITHMS);
+    assert_int_equal(spdm_response->other_params_selection, SPDM_ALGORITHMS_OPAQUE_DATA_FORMAT_1);
+
+    /* Sub Case 2: other_params_support added version 1.3 data in version 1.2*/
+    m_libspdm_negotiate_algorithm_request18.spdm_request_version10.other_params_support =
+        SPDM_ALGORITHMS_OPAQUE_DATA_FORMAT_1 |
+        SPDM_ALGORITHMS_MULTI_KEY_CONN;
+    spdm_context->local_context.algorithm.other_params_support =
+        SPDM_ALGORITHMS_OPAQUE_DATA_FORMAT_1 |
+        SPDM_ALGORITHMS_MULTI_KEY_CONN;
```

Comment:
for version 1.2, I think here `spdm_context->local_context.algorithm.other_params_support` for responder should not support `SPDM_ALGORITHMS_MULTI_KEY_CONN`.